content/articles/build-a-weather-forecast-app-with-next-js/index.md

```diff
@@ -0,0 +1,539 @@
+
+
+### Performance
+
+Because of static sites and server-side rendering, Next.js apps are lightning-fast. They work by default, thanks to a slew of performance-enhancing technologies like Image Optimization, which debuted in Next 10.
+
+As a result, if you use Next.js, you'll get automatic server rendering and code splitting (which will improve performance). Furthermore, SSR (Server Side Rendering) will significantly increase your application's performance.
+
+When it comes to React, there are a few things that rule it out of the discussion. It only supports client-side rendering out of the box, which is insufficient if you want to construct a high-performance application.
+
+
+### Features
+
+Next.js makes use of React to assist developers in creating single-page apps.
+
+It has the following features:
+
+
+
+- Rendering on the server (SSR)
+- Exporting static data (SSG)
+- Pre-rendering
+- Build size optimization is done automatically.
+- Compilation of code more quickly
+
+All of these capabilities will assist you in developing a functional and ready-to-use application.
+
+With frameworks like Redux, React can simply be extended to integrate functionality like routing and state management patterns. React is a simple framework that can be adapted for almost any project.
+
+
+### Documentation
+
+This is self-evident, but it is frequently overlooked. While the stylish home pages of certain frameworks may capture your eye, you'll still need additional courses, books, tutorials, and articles to get started, in addition to the monotonous and dry documentation.
+
+In any software project, effective documentation may make it much easier to find out how to use the tools, which libraries to utilize, and so on. Both React and Next.js provide amazing documentation options.
+
+Next.js includes a set of "learn-by-doing" documentation that leads you through tasks like component development and routing. React provides a similar setup, with multiple tutorials covering the fundamentals.
+
+Check to browse the official React documentation to learn more about React at a high level. We also recommend reading the official Next.js documentation to obtain a better understanding of the framework.
+
+In my opinion, next js only scores high in performance than React all others are even.
+
+Now that we’ve taken a look at Next js and why you should use it, let’s move on to build a weather forecast application in Next .
+
+
+### Requirements
+
+
+### To get started you just need a basic understanding of javascript and react knowledge and also signup to openweathermap to make use of their API, it’s completely free. so head over to [https://openweathermap.org/api](https://openweathermap.org/api) and signup. Click on the API key, you should see something like this:
+![api_key](build-a-weather-forecast-app-with-next-js/api_key.PNG)
```

Comment:
The image path is incorrect. Please refer the guidelines or have a look at other files.